code/modules/assembly/doorcontrol.dm

```diff
@@ -152,3 +152,17 @@
 			C.cremate(usr)
 
 	addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 50)
+
+/obj/item/assembly/control/shieldwallgen
+	name = "holofield controller"
+	desc = "A small device used to remotely operate holofield generators."
+
+/obj/item/assembly/control/shieldwallgen/activate()
```

Comment:
on the /obj/machinery/power/shieldwallgen you do `id = "[REF(port)][id]"` if on shuttle, but you don't do the same with the button causing it to not work

Reply:
addressed in 0a3c0c1